src/tasks/phase2/singular_value_decomposition.py

```diff
@@ -0,0 +1,78 @@
+from database_connection import DatabaseConnection
+from histogram_of_gradients import HistogramOfGradients
+import pprint
+import numpy as np
+from utils import visualize_images,plot_scree_test
+import matplotlib
+import matplotlib.pyplot as plt
+from sklearn.decomposition import TruncatedSVD
+
+
+# Is a temperarory class design
+class SingularValueDecomposition:
+
+    def __init__(self):
+        self.dbconnection = DatabaseConnection()
+        pass
+
+    def get_latent_semantics(self, data_matrix, n_components):
+        u, s, vt = np.linalg.svd(data_matrix,full_matrices=False)
+        u = np.matrix(u[:,:n_components])
+        s = np.diag(s[:n_components])
+        vt = np.matrix(vt[:n_components,:])
+        # plot_scree_test(np.diagonal(s))
+        return u, s, vt
+    
+    def svd_hog(self,k,input_image):
```

Comment:
Check with Suryanshu if this function is required

Reply:
we can remove everything except get_latent_semantics. I think plot scree test would have been  moved to utils